postgres/src/test/java/org/jdbi/v3/postgres/TestInetAddressPg.java

```diff
@@ -13,19 +13,72 @@
  */
 package org.jdbi.v3.postgres;
 
+import java.net.Inet4Address;
+import java.net.InetAddress;
+
 import de.softwareforge.testing.postgres.junit5.EmbeddedPgExtension;
 import de.softwareforge.testing.postgres.junit5.MultiDatabaseBuilder;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Sql;
 import org.jdbi.v3.core.argument.TestInetAddressH2;
 import org.jdbi.v3.core.junit5.PgDatabaseExtension;
+import org.jdbi.v3.testing.junit5.JdbiExtension;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class TestInetAddressPg extends TestInetAddressH2 { { dbExtension = PgDatabaseExtension.instance(pg).withPlugin(new PostgresPlugin()); }
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestInetAddressPg extends TestInetAddressH2 {
+
+    {
+        dbExtension = PgDatabaseExtension.instance(pg).withPlugin(new PostgresPlugin());
+    }
 
     @RegisterExtension
     public static EmbeddedPgExtension pg = MultiDatabaseBuilder.instanceWithDefaults().build();
 
+    @RegisterExtension
+    public JdbiExtension pgExtension = JdbiExtension.postgres(pg).withPlugin(new PostgresPlugin());
+
     @Override
     protected String getInetType() {
         return "INET";
     }
+
+    @Test
+    public void testPgInetAddress() throws Exception {
+        try (Handle h = dbExtension.getJdbi().open()) {
+            h.execute("CREATE TABLE inet_test (ipaddress " + getInetType() + ")");
+
+            InetAddress inetAddrIn = InetAddress.getByName("8.8.8.8");
+            assertThat(inetAddrIn).isInstanceOf(Inet4Address.class);
+
+            // insert IP address into column of type INET
+            int update1Count = h.createUpdate(Sql.of("INSERT INTO inet_test (ipaddress) VALUES (:address)"))
+                .bind("address", inetAddrIn)
+                .execute();
+            assertThat(update1Count).isOne();
+
+            // read back the inserted IP address
+            InetAddress inetAddr1Out = h.createQuery(Sql.of("SELECT ipaddress FROM inet_test"))
+                .mapTo(InetAddress.class)
```

Comment:
this is the big difference between this unit test and the bug report. You are doing this right. I am pretty sure that your test will work even without the mapper registered.

Reply:
makes sense now